### PR TITLE
Updates by @vsoch

### PR DESCRIPTION
**Pull request by vsoch**
This is a (work in progress) pull request to show progress on the draft. I'm going to first go through the bullets and then (if appropriate) write them into more human friendly sentences.

Signed-off-by: Vanessa Sochat <vsochat@stanford.edu>

### CODE REVIEW

**Comment by vsoch** [comment]
hey @nuest ! I'm all done with my changes, specifically I've taken all bullet points and flushed them out into full fledged lines. As instructed, when a reference was needed I added a link or REF. This should be ready to merge, I won't be doing any further work on it, at least until there is a more finalized draft to look at.

I generally maintained all rules / structure, with the exception of Rule 8 (add the build/run to comments at end) which was redundant from a previous rule to suggest using comments for transparency, etc. What I did is move content from there into that issue, and modified rule 8 to be about creating a protocol for starting a new project (which was missing and I think important).

Please merge and tweak to your needs! I'm moving cross country in the next two weeks plus a conference, so I wanted to get this finished for you today. Cheers!

**Comment by vsoch** [comment]
ping @nuest - did you see this?

**Comment by nuest** [comment]
Yes @vsoch - thank you for the reminder! A deadline took control of my last two weeks, and I am traveling now, but will take a look at it by the end of the week. I can already say: Wow! :open_mouth: 

There's also plenty of discussion issues to follow up on - this manuscript is my reward for the other submitted one ;-) and I look forward to reading your work.

**Comment by nuest** [comment]
Hi Vanessa - thanks for pushing the manuscript a _huge_ step forward!

I'm working through your changes now, will create a new PR for you and the others to take a look at soon.

**Comment by nuest** [comment]
Sorry, I just realised that my pushes go of course directly to master. So, first changes after this PR are in https://github.com/nuest/ten-simple-rules-dockerfiles/commit/4a87e3e3ad43feacd98722f1521e500191bb17bb - will update here tomorrow with more.

File: ten-simple-rules-dockerfiles.Rmd [hunk 1]
```diff
@@ -34,57 +35,57 @@ csl: plos.csl
...
-Not everybody needs to understand and write Dockerfiles, as is reflected in rule 1.
-However, as one can argue that _"the number of unique reserach environments approximates the number of researchers"_ [@nust_opening_2017], the researchers who do need to craft their own environment definition should follow practices taking into account the scientific context.
+With access to version control systems [REF GITHUB, GITLAB, etc.] it has become increasingly easy to not only share algorithms, but also instructions for building and testing.
+Within this content, it is suggested to include instructions for building containers to package software and data dependencies [@boettiger_introduction_2017].
+By way of providing these instructions along with thorough documentation, it's much more likely to be able to reproduce an analysis workflow.
+A container is a portabal, encapsulated environment that is built from human and machine readable instructions.
```
Comment (psychemedia) [inline ten-simple-rules-dockerfiles.Rmd]: typo: portabal -> portable

Comment (vsoch) [reply ten-simple-rules-dockerfiles.Rmd]: Thanks! This was already merged so perhaps we can check for it in master?

File: ten-simple-rules-dockerfiles.Rmd [hunk 2]
```diff
@@ -34,57 +35,57 @@ csl: plos.csl
...
+For example, containers have been created to host scientific notebooks [@rule_ten_2019], and share reproducible workflows (cf. Rule 10 of @sandve_ten_2013).
+<!-- the virtual laboratory reference isn't great, I removed it-->
+
+While there are several tutorials for using containers for reproducible research [^1], there is no extensive examination for how to write the actual instructions to create the containers.
```
Comment (psychemedia) [inline ten-simple-rules-dockerfiles.Rmd]: Would it make sense to try to make bridges to related communities in eg devops; so maybe things like https://dl.acm.org/citation.cfm?id=3240299 or https://dl.acm.org/citation.cfm?id=2723882 ? A search around phrases like "literate devops" might also turn up emerging practice in devops that is inspired by some of the literate/reproducible practice in data analysis and reproducible research?

File: ten-simple-rules-dockerfiles.Rmd (hunk 1 above)
Comment (vsoch) [reply ten-simple-rules-dockerfiles.Rmd]: Here you go: https://github.com/nuest/ten-simple-rules-dockerfiles/pull/25

For future edits, please open PR, issue or similar (we shouldn't be commenting on closed and merged PR, as @nuest is likely to have already made changes).

File: ten-simple-rules-dockerfiles.Rmd (hunk 2 above)
Comment (vsoch) [reply ten-simple-rules-dockerfiles.Rmd]: Please see https://github.com/nuest/ten-simple-rules-dockerfiles/pull/21#pullrequestreview-327830486.

File: ten-simple-rules-dockerfiles.Rmd [hunk 3]
```diff
@@ -34,57 +35,57 @@ csl: plos.csl
...
+Several platforms for facilitating reproducible research are built on top of containers [@brinckman_computing_2018; @code_ocean_2019; @simko_reana_2019; @jupyter_binder_2018; @nust_opening_2017], but they hide any complexity from the researcher.
+While not everybody needs to understand and write Dockerfiles, as _"the number of unique research environments approximates the number of researchers"_ [@nust_opening_2017], the researchers who do need to craft their own environment definition should follow good practices.
```
Comment (psychemedia) [inline ten-simple-rules-dockerfiles.Rmd]: There is also scope for folk being able to "read and modify" pre-existing Dockerfiles. Good practice in writing eg using conventions, makes things easier to read; pragmatically, for folk wishing to make changes, sticking to conventions can also help with the writing/modification even if they aren't sure how things actually work...

Comment (nuest) [reply ten-simple-rules-dockerfiles.Rmd]: I agree with this point and will take a look again if it is mentioned clearly enough.

Comment (nuest) [reply ten-simple-rules-dockerfiles.Rmd]: Tried to clarify the advantage of conventions in this change to the introduction: https://github.com/nuest/ten-simple-rules-dockerfiles/commit/b83bc16a08d90d609c4d74c1ac59b5a58eba6077

**Pull request merged**